### ISSUE TITLE
scExpression: implement developmental filter and grouping

### ISSUE THREAD

**Issue by signechambers1** [chanzuckerberg/single-cell-data-portal#2339]
Implement developmental filter and grouping

**Comment by tihuan** [comment]
@signechambers1 do we know who's gonna help figure out the developmental group mapping for FE to implement? Will that be Pablo or? Thank you!

**Comment by signechambers1** [comment]
I think we want to use the same one portal is using, @pablo-gar do you agree? Are there any changes you think we'd need to make?

**Comment by tihuan** [comment]
Ahh I see!

Something like this?

Homo sapiens: https://github.com/chanzuckerberg/single-cell-data-portal/blob/b68bfc59888b7e82c8a82c20d50dfd8ec276b70a/frontend/src/components/common/Filter/common/entities.ts#L100

Mouse: https://github.com/chanzuckerberg/single-cell-data-portal/blob/b68bfc59888b7e82c8a82c20d50dfd8ec276b70a/frontend/src/components/common/Filter/common/entities.ts#L184

Thank you!

**Comment by tihuan** [comment]
Hmm actually what Data Portal has doesn't seem to be enough to cover the development stages returned by the BE. For example, Data Portal currently has `Mature (13+ years)` with the following labels:

```ts
{
      label: "Mature (13+ years)",
      ontology_term_id: "HsapDv:0000204",
      children: [
        {
          label: "Adolescent (13–19 years)",
          ontology_term_id: "HsapDv:0000086",
        },
        {
          label: "Human adult (19+ years)",
          ontology_term_id: "HsapDv:0000087",
          children: [
            {
              label: "Early adulthood (19–45 years)",
              ontology_term_id: "HsapDv:0000088",
            },
            {
              label: "Late adulthood (45+ years)",
              ontology_term_id: "HsapDv:0000091",
            },
          ],
        },
      ],
    },
```

But BE actually returns very granular labels, like `{HsapDv:0000140: "46-year-old human stage"}`, `{HsapDv:0000242: "eighth decade human stage"}`, `{HsapDv:0000192: "19-month-old human stage"}`, which are not available in the hard coded mapping in Data Portal right now as shown above.

So in order to have the right hierarchical mapping, we might need to get the full human and mouse development stage hierarchical mappings from somewhere?

CC: @pablo-gar @signechambers1 

Thank you!

**Comment by atolopko-czi** [comment]
@tihuan @MDunitz In case we find backend support is needed to implement this properly, note that I started (and then tabled) a naive attempt at implementing developmental stage hierarchy support in #2052. We may be able to leverage some of that work if needed.

**Comment by tihuan** [comment]
Oh amazing! Thanks so much for all the info, @brianraymor and @atolopko-czi 🙏 Let me read the docs now!

**Comment by tihuan** [comment]
Ah interesting! Looking at this [human stage tree](https://github.com/obophenotype/developmental-stage-ontologies/blob/master/external/bgee/report.md#homo-sapiens) by BGEE, it seems to cover most of the stages except for the "decade" ones. Like, `{HsapDv:0000237: "third decade human stage"}`, `{HsapDv:0000238: "fourth decade human stage"}`, etc.. 

Is that because those decade ones aren't part of BGEE? If that's the only exception, we can just manually add those "decade" ones in the [BGEE tree](https://github.com/obophenotype/developmental-stage-ontologies/blob/master/external/bgee/report.md#homo-sapiens)?

![Screen Shot 2022-04-26 at 11.35.43 AM.png](https://images.zenhubusercontent.com/5ebed8780d2d150fc281ad55/1533a582-da3d-4c09-830d-f871bf7936c9)

**Comment by tihuan** [comment]
Also `{ label: "Mature (13+ years)", ontology_term_id: "HsapDv:0000204" }` in our hard coded list above doesn't seem to exist in the [BGEE tree](https://github.com/obophenotype/developmental-stage-ontologies/blob/master/external/bgee/report.md#homo-sapiens), so maybe that's part of the "decade" ones?

**Comment by brianraymor** [comment]
We started with BGEE approach and then simplified further. The epic documents the final set that Jason helped us to select.

**Comment by tihuan** [comment]
Ah got it! All makes sense now 🎉 Thanks so much, Brian!

**Comment by tihuan** [comment]
@signechambers1 Adding P2 to this ticket since it's blocked by BE as a stretch goal. Thank you!

**Comment by signechambers1** [comment]
thanks @tihuan! I'm going to remove it from the Q2 epic since we are not prioritizing this for this quarter
